File: backend/src/main/java/com/jbconnect/healthclinix/entity/user/facade/UserCrudFacade.java
package com.jbconnect.healthclinix.entity.user.facade;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.jbconnect.healthclinix.entity.user.dto.UserRequestDTO;
import com.jbconnect.healthclinix.entity.user.dto.UserResponseDTO;

public interface UserCrudFacade {

	UserResponseDTO save(UserRequestDTO userRequestDTO);

	UserResponseDTO findOne(Long id);

	Page<UserResponseDTO> getByFilters(Pageable pageable, UserRequestDTO userRequestDTO);

}

File: backendkotlin/src/main/kotlin/com/jbconnect/healthclinix/healthclinix/entity/user/facade/UserCrudFacade.kt
package com.jbconnect.healthclinix.healthclinix.entity.user.facade

import com.jbconnect.healthclinix.healthclinix.entity.user.dto.UserRequestDTO
import com.jbconnect.healthclinix.healthclinix.entity.user.dto.UserResponseDTO
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable


interface UserCrudFacade {
    fun save(userRequestDTO: UserRequestDTO) : UserResponseDTO

    fun getByFilters(pageable: Pageable, userRequestDTO: UserRequestDTO): Page<UserResponseDTO>
}